pkg/controller/atlasdeployment/serverless_deployment.go

```diff
@@ -54,3 +77,16 @@ func ensureServerlessInstanceState(ctx *workflow.Context, project *mdbv1.AtlasPr
 		return atlasDeployment, workflow.Terminate(workflow.Internal, fmt.Sprintf("unknown deployment state %q", atlasDeployment.StateName))
 	}
 }
+
+func isEqual(a *mongodbatlas.Cluster, c *mongodbatlas.Cluster) bool {
```

Comment:
I'd rename as `isEqualTags()` because we are only checking the tags, not the cluster objects in full.

Alternatively, it can be left as isEqual() but just passing 2 arrays of tags to be compared.